tests/unit/ForeignJurisdiction.spec.ts

```diff
@@ -0,0 +1,104 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { mount } from '@vue/test-utils'
+import { ForeignJurisdiction } from '@/components/common'
+
+Vue.use(Vuetify)
+
+const vuetify = new Vuetify({})
+
+describe('ForeignJurisdiction', () => {
+  it('initializes correctly', () => {
+    const wrapper = mount(ForeignJurisdiction,
+      {
+        vuetify
+      })
+    const vm: any = wrapper.vm
+
+    expect(vm.selectedCountry).toBe('')
+    expect(vm.selectedRegion).toBe('')
+    expect(vm.countryNames.length).toBe(249)
+    expect(wrapper.emitted('valid')).toBeFalsy()
+    wrapper.destroy()
+  })
+
+  it('set correct country when selected via the country selector', async () => {
+    const wrapper = mount(ForeignJurisdiction,
+      {
+        vuetify
+      })
+    const vm: any = wrapper.vm
+
+    const countriesSelector = wrapper.find('#country-selector')
+    await countriesSelector.setValue('Lebanon')
+
+    expect(vm.selectedCountry).toBe('Lebanon')
+    expect(vm.selectedRegion).toBe('')
+    wrapper.destroy()
+  })
+
+  it('set correct region when selected via the region selector', async () => {
+    const wrapper = mount(ForeignJurisdiction,
+      {
+        vuetify
+      })
+    const vm: any = wrapper.vm
+
+    const countriesSelector = wrapper.find('#country-selector')
+    await countriesSelector.setValue('Canada')
+    const regionsSelector = wrapper.find('#region-selector')
+    await regionsSelector.setValue('Ontario')
+
+    expect(vm.selectedCountry).toBe('Canada')
+    expect(vm.selectedRegion).toBe('Ontario')
+    wrapper.destroy()
+  })
+
+  it('emit correct validation when country is selected', async () => {
+    const wrapper = mount(ForeignJurisdiction,
+      {
+        vuetify
+      })
+    const vm: any = wrapper.vm
+
+    const countriesSelector = wrapper.find('#country-selector')
+    await countriesSelector.setValue('Lebanon')
+
+    vm.emitChangedCountry()
```

Comment:
I think you can actually emit a change event on the component, so that you don't have to call this directly.

Reply:
Example:
https://github.com/bcgov/business-edit-ui/blob/e37407b44e3233773cfb913151f0bc3812bde0fc/tests/unit/OrgPerson.spec.ts#L279

---

To do this on some components, you have to select them by either id or classname -- Vuetify uses one of those on the component's div and the other on the embedded HTML input element.

Example showing the id on the input element, but if there was a class then it would be on the top div:

![image](https://github.com/bcgov/business-filings-ui/assets/10002889/472de5cf-4188-41b0-b686-353d9ebcebc3)

Reply:
Example:
https://github.com/bcgov/business-edit-ui/blob/e37407b44e3233773cfb913151f0bc3812bde0fc/tests/unit/OrgPerson.spec.ts#L279

---

To do this on some components, you have to select them by either id or classname -- Vuetify uses one of those on the component's div and the other on the embedded HTML input element.

Example showing the id on the input element, but if there was a class then it would be on the top div:

![image](https://github.com/bcgov/business-filings-ui/assets/10002889/41d6bc1e-3309-4d12-b44a-bff944d6264d)